From 605f433fd4cf3de251b33847d763a0f7ff94fd40 Mon Sep 17 00:00:00 2001
From: Kari Argillander <kari.argillander@gmail.com>
Date: Sun, 29 Aug 2021 17:42:39 +0300
Subject: [PATCH 35/77] fs/ntfs3: Change how module init/info messages are
 displayed

Usually in file system init() messages are only displayed in info level.
Change level from notice to info, but keep CONFIG_NTFS3_64BIT_CLUSTER in
notice level. Also this need even more attention so let's put big
warning here so that nobody will not try accidentally use it.

There is also no good reason to display internal stuff like binary tree
search. This is always on option which can only disabled for debugging
purposes by developer. Also this message does not even check if
developer has disabled it or not so it is useless info.

Signed-off-by: Kari Argillander <kari.argillander@gmail.com>
Signed-off-by: Konstantin Komarov <almaz.alexandrovich@paragon-software.com>
---
 fs/ntfs3/super.c | 27 ++++++++-------------------
 1 file changed, 8 insertions(+), 19 deletions(-)

diff --git a/fs/ntfs3/super.c b/fs/ntfs3/super.c
index dbecf095d..55bbc9200 100644
--- a/fs/ntfs3/super.c
+++ b/fs/ntfs3/super.c
@@ -1448,21 +1448,14 @@ static int __init init_ntfs_fs(void)
 {
 	int err;
 
-	pr_notice("ntfs3: Index binary search\n");
-	pr_notice("ntfs3: Hot fix free clusters\n");
-	pr_notice("ntfs3: Max link count %u\n", NTFS_LINK_MAX);
+	pr_info("ntfs3: Max link count %u\n", NTFS_LINK_MAX);
 
-#ifdef CONFIG_NTFS3_FS_POSIX_ACL
-	pr_notice("ntfs3: Enabled Linux POSIX ACLs support\n");
-#endif
-#ifdef CONFIG_NTFS3_64BIT_CLUSTER
-	pr_notice("ntfs3: Activated 64 bits per cluster\n");
-#else
-	pr_notice("ntfs3: Activated 32 bits per cluster\n");
-#endif
-#ifdef CONFIG_NTFS3_LZX_XPRESS
-	pr_notice("ntfs3: Read-only LZX/Xpress compression included\n");
-#endif
+	if (IS_ENABLED(CONFIG_NTFS3_FS_POSIX_ACL))
+		pr_info("ntfs3: Enabled Linux POSIX ACLs support\n");
+	if (IS_ENABLED(CONFIG_NTFS3_64BIT_CLUSTER))
+		pr_notice("ntfs3: Warning: Activated 64 bits per cluster. Windows does not support this\n");
+	if (IS_ENABLED(CONFIG_NTFS3_LZX_XPRESS))
+		pr_info("ntfs3: Read-only LZX/Xpress compression included\n");
 
 	err = ntfs3_init_bitmap();
 	if (err)
@@ -1502,15 +1495,11 @@ static void __exit exit_ntfs_fs(void)
 
 MODULE_LICENSE("GPL");
 MODULE_DESCRIPTION("ntfs3 read/write filesystem");
-MODULE_INFO(behaviour, "Index binary search");
-MODULE_INFO(behaviour, "Hot fix free clusters");
 #ifdef CONFIG_NTFS3_FS_POSIX_ACL
 MODULE_INFO(behaviour, "Enabled Linux POSIX ACLs support");
 #endif
 #ifdef CONFIG_NTFS3_64BIT_CLUSTER
-MODULE_INFO(cluster, "Activated 64 bits per cluster");
-#else
-MODULE_INFO(cluster, "Activated 32 bits per cluster");
+MODULE_INFO(cluster, "Warning: Activated 64 bits per cluster. Windows does not support this");
 #endif
 #ifdef CONFIG_NTFS3_LZX_XPRESS
 MODULE_INFO(compression, "Read-only lzx/xpress compression included");
-- 
2.33.0.328.g8b7c11b866

